pkg/webhooks/validation/controller/controller.go

```diff
@@ -315,10 +336,16 @@ func (c *Controller) processNextWorkItem() (cont bool) {
 	}
 
 	// return false when leader lost in case go routine leak.
-	if req.description == QuitSignal {
+	if req.event == quitEvent {
 		c.queue.Forget(req)
 		return false
 	}
+	// when the validation webhook is updated, likely because defaultRevision was
```

Comment:
One concern is that we essentially have a downtime where invalid configs may get through. I am not sure if we can avoid this easily though

Reply:
Why ? Doesn't Reconcile just update the public key ?

Reply:
And I think last discussion was to have Istioctl take care of this, so we don't have to have such high privs in istiod

Reply:
on initial install we mark it fail open, until istiod is ready. it was to fix some races where we apply istio resources during install.

We could probably remove it if we finally get rid of the envoyfilters for telemetry.

Reply:
> > How about a more extreme approach - document the URL ( istiod-rev.istio-system.svc/validate ) and provide an example .yaml, and have users apply the yaml directly or with istioctl ?
> 
> I'm not sure that this is aligned with our goal of making revisions easier to use. Recent work like the revision tags proposal were aimed at making revisions easier to adopt, I think a manual step to make revisions work would be a step backwards there

I didn't say it has to be a manual step performed by the user - in fact none of the revision upgrade should be manually done in the end. Users will have either CI/CD or use istioctl to do all this. 

The problem is that right now all this risky and complicated thing is hidden in helm - so user can't automate this easily
 with the tools they use for all the other k8s management they are doing. That may be fine for istio-system - but I don't think security sensitive stuff should be so obfuscated. 

> 
> > not sure it's aligned with the goal of reducing privs
> 
> That's true, but I think we're married to webhook privileges for the near future (and we don't have a design yet for how the non-patching model would work with helm). Agreed we need to reduce istiod privs long-term but I don't know that it should be a blocker for this PR which is itself a short-term fix for broken revisions

There is another short term fix - just create a istiod.istio-system service that points to the right revision if default is not installed.